### PR TITLE
Turn on strict type predicate

### PR DESCRIPTION
**Pull request by GabeDeBacker**
Starts to address issue #242. Turns on "strict type predicate" which performs linting operations such as verifying that "if" conditions for "undefined" or against "types" do not evaluate to constant true or false expressions.

### CODE REVIEW

File: src/fileMapper.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ export class FileMapper implements Disposable {
             this.addToFileMapping(Utilities.getFsPathWithFragment(origUri), uri);
             this.saveBasePath(origUri, uri, uriBase);
             this.fileRemapping.forEach((value: Uri | undefined, key: string) => {
-                if (value === null) {
+                if (value) {
```
Comment (roywill) [inline src/fileMapper.ts:140]: Inverted?

**Comment by roywill** [comment]
>             if (!this.groupBy) {

Seems like a waste of time.

---
Refers to: src/resultsListController.ts:439 in 4419c9c. [](commit_id = 4419c9ca92004ac3500f4f9e15e9a046108edbe3, deletion_comment = False)

File: src/resultsListController.ts [hunk 2]
```diff
@@ -440,7 +440,11 @@ export class ResultsListController implements Disposable {
                 continue;
             }
```
Comment (GabeDeBacker) [inline src/resultsListController.ts:441]: This condition is always false as well

File: src/resultsListController.ts [hunk 3]
```diff
@@ -480,26 +484,29 @@ export class ResultsListController implements Disposable {
                     valueB = a[this.sortBy.column];
                 }
 
-                if (valueA === undefined || valueA.value === undefined) {
-                    comp = -1;
-                    if (valueB === undefined || valueB.value === undefined) {
+                if (valueA === undefined ||
```
Comment (roywill) [inline src/resultsListController.ts:487]: Pull this into an equality function.

Comment (GabeDeBacker) [reply src/resultsListController.ts]: Doing so would remove the TypeScript compiler's ability to know they are not undefined on line 498.

I've made this code easier to understand though.

---
In reply to: [408282753](https://github.com/microsoft/sarif-vscode-extension/pull/243#discussion_r408282753) [](ancestors = 408282753)

**Pull request merged**